### PR TITLE
Add str_to_assoc, and document multiple names.

### PR DESCRIPTION
**Pull request by rosiel**
**GitHub Issue**: (link)

* Other Relevant Links: we've been trying to do migrations with the MIG. 

# What does this Pull Request do?

Provides the custom process plugin for turning a CSV into structured data, e.g. if you have multiple Linked Agents, you can import them all! You can even import them with different relators!.

It also makes the migration use this method, and explains it in documentation.
 
# What's new?

migrate_islandora_csv now contains a Migrate process plugin. Oh no, it's not a pure Feature anymore! (is that ok?) The plugin, as explained in the TUTORIAL, takes 
`['a','b','c']` along with a key ('my_key') and turns it into:

```php
[
  ['my_key' => 'a'], 
  ['my_key' => 'b'],
  ['my_key' => 'c'],
]
```
The plugin is really generic, and may be of broader interest.

The sample data now includes another column with multiple names (Clawde and Clawbster for all values) to demonstrate how this works. 

The node migration and the tutorial show how to create an associative array from a csv column, to use in complex fields.

# How should this be tested?

* pull these changes
* `drush cr` (to make the process plugin available)
* `drush fim -y migrate_islandora_csv
* [reset/roll back the Node migration in case you've already run it - `drush mrs node; drush mr node;`]
* `drush mim node` to run the node migration.

Validate:
* Are there two new terms in the Corporate Body vocabulary at http://localhost:8000/admin/structure/taxonomy/manage/corporate_body/overview ?
* Do the generated nodes show the Photographer (pht) and Provider (prv)s?

note: when I import, i get a bunch of warnings about indexing the Default Solr content index. They aren't caused by these changes! they look like:
` [warning] Warning: While indexing items on search index Default Solr content index, 1 item(s) did not have a view mode configured for one or more "Rendered item" fields.` and it may be because of a mismatch in age of various stack components... 

# Interested parties
Yo @Islandora/8-x-committers, shoutout to @dannylamb and @seth-shaw-unlv, @mbolam,  and @rtilla1.

### CODE REVIEW

File: TUTORIAL.md [hunk 1]
```diff
@@ -30,7 +30,9 @@ This repository is also a Drupal Feature that, when enabled as a module, will cr
 
 This repository also contains a `data` folder containing a CSV and sample images, as a convenience so that the accompanying files are easily available on the Drupal server running the migration. (This is not the recommended method for making files available to Drupal in a real migration.)
 
-When you are ready to create your actual migrations, this repository can function as a template for you to create the yml files defining your own migrations.
+And, at the moment, this repository contains a Migrate process plugin that transforms strings into associative arrays. This is useful for populating multiple Linked Agent fields. It will be available when this module is enabled, and the node migration uses it. It was written generically and will hpefully become part of Migrate Plus, but for now it is here.
```
[the remark this answers is not in the document]
Comment (rosiel) [reply TUTORIAL.md]: ah! thanks.

**Comment by seth-shaw-unlv** [comment]
I haven't tested it yet, but it looks good. I wouldn't worry about the plugin causing the module to not be strictly config.

**Comment by seth-shaw-unlv** [comment]
Oh, I should also note that the `[warning] Warning: While indexing items...` warnings are there because we haven't gotten search_api_solr indexing configurations in defaults. See https://github.com/Islandora/documentation/issues/1154.

**Comment by elizoller** [comment]
I pulled down this PR into a running islandora 8 instance, uninstalled module, enabled module, ran the migrations and received linked agents with different relators (as designed i think?)
Screen Shot 2019-11-07 at 10 51 01 AM

**Pull request merged**